core/src/Streamly/Internal/Data/Array/Foreign/Mut/Type.hs

```diff
@@ -2344,22 +2377,14 @@ strip eq arr@Array{..} = do
 {-# INLINE bubbleAsc #-}
 bubbleAsc :: (MonadIO m, Storable a) =>
     (a -> a -> Ordering) -> Array a -> m ()
-bubbleAsc cmp0 arr =
-    when (l > 1) $ do
-        x <- getIndexUnsafe (l - 1) arr
-        go x (l - 2)
-
-        where
-
-        l = length arr
+bubbleAsc = bubbleWith True
 
-        go x i =
-            if i >= 0
-            then do
-                x1 <- getIndexUnsafe i arr
-                case x1 `cmp0` x of
-                    GT -> do
-                        putIndexUnsafe (i + 1) x1 arr
-                        go x (i - 1)
-                    _ -> putIndexUnsafe (i + 1) x arr
-            else putIndexUnsafe (i + 1) x arr
+-- | Given an array sorted in descending order except the last element being out
+-- of order, use bubble sort to place the last element at the right place such
+-- that the array remains sorted in descending order.
+--
+-- /Pre-release/
+{-# INLINE bubbleDesc #-}
+bubbleDesc :: (MonadIO m, Storable a) =>
+    (a -> a -> Ordering) -> Array a -> m ()
+bubbleDesc = bubbleWith False
```

Comment:
1. I guess we do not need `bubbleWith` and `bubbleDesc`. We can just use `bubbleAsc` with `flip compare` to perform a descending sort?
2. If so we can rename `bubbleAsc` to just `bubble`, it will implicitly mean ascending sort order.